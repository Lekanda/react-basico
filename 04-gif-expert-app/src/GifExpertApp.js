import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
// import ReactDOM from 'react-dom';

export const GifExpertApp = ({defaultCategories=[]}) => {

    // const [categories, setcategories] = useState(['One Punch']);
    const [categories, setcategories] = useState(defaultCategories);
    

    return (
        <>
            <h2>GifExpertApp</h2>
            <hr/>
            <AddCategory setcategories={setcategories}/>
            <hr/>
            <ol>
                {  
                    categories.map( (category => 
                            <GifGrid 
                                key={ category }
                                category={ category }
                            />
                    )) 
                }
                
            </ol>
        </>
    )
}

export default GifExpertApp
