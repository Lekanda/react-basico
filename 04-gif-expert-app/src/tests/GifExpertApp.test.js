import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Tests en GifExpertApp.js', ()=>{

    test('Debe comparar Snapshot', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
})