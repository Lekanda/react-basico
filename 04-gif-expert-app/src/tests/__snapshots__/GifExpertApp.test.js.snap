// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests en GifExpertApp.js Debe comparar Snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Tests en GifExpertApp.js Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Samurai X"
      key="Samurai X"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
